### PR TITLE
Feat: add threading config for email receiver in Alertmanager secret config

### PR DESCRIPTION
**Pull request by nutmos**
## Description

This PR adds threading support for email receiver in Alertmanager secret config.

Threading is supported from AM 0.30.0 onwards.

https://github.com/prometheus/alertmanager/releases/tag/v0.30.0

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification
Unit Testing

## Changelog entry

```release-note
- Add threading support for email receiver in Alertmanager secret config
```

### CODE REVIEW

File: pkg/alertmanager/amcfg_test.go [hunk 1]
```diff
@@ -5855,6 +5855,44 @@ func TestSanitizeEmailConfig(t *testing.T) {
...
+								Threading: &emailThreadingConfig{
+									Enabled:      ptr.To(true),
+									ThreadByDate: "daily",
+								},
+							},
+						},
+					},
+				},
+			},
+			golden: "test_threading_is_added_in_email_config_for_supported_versions.golden",
```
Comment (heliapb) [inline pkg/alertmanager/amcfg_test.go:6185]: Hi @nutmos I was looking into the alertmanager change, could you add a test as well for the empty ThreadByDate as well since is the default. Thanks

**Comment by nutmos** [comment]
@heliapb Please check

**Comment by nutmos** [comment]
@simonpasquier Updated. Please check again.

**Comment by nutmos** [comment]
@heliapb @simonpasquier Please review again. Thanks!

File: pkg/alertmanager/amcfg.go [hunk 2]
```diff
@@ -2584,6 +2584,18 @@ func (ec *emailConfig) sanitize(amVersion semver.Version, logger *slog.Logger) e
...
 	}
 
+	if ec.Threading != nil && amVersion.LT(semver.MustParse("0.30.0")) {
+		msg := "'threading' supported in Alertmanager >= 0.30.0 only - dropping field from provided config"
+		logger.Warn(msg, "current_version", amVersion.String())
+		ec.Threading = nil
+	}
+
+	if t := ec.Threading; t != nil {
+		if t.ThreadByDate != "" && t.ThreadByDate != "daily" && t.ThreadByDate != "none" {
```
Comment (simonpasquier) [inline pkg/alertmanager/amcfg.go]: empty thread_by_date would trigger a config error

**Comment by nutmos** [comment]
@simonpasquier Could you please have a look?
